frontend/src/metabase/components/FormCategoryInput/FormCategoryInput.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { forwardRef, ReactNode, Ref, useCallback } from "react";
+import { useField } from "formik";
+import { useUniqueId } from "metabase/hooks/use-unique-id";
+import FormField from "metabase/core/components/FormField";
+
+import type Field from "metabase-lib/metadata/Field";
+
+import CategoryFieldPicker from "./CategoryFieldPicker";
+
+export interface FormCategoryInputProps {
+  name: string;
+  title?: string;
+  description?: ReactNode;
+  nullable?: boolean;
+  className?: string;
+  style?: React.CSSProperties;
+  fieldInstance: Field;
```

Comment:
nit: why it's `fieldInstance` and not a `field`? Mostly out of curiosity.

Reply:
I think the name comes from `redux-form` / `FormikForm` times when the `field` prop used to be a field definition object (with `name`, `title`, etc.). Think we'd be good calling it `field` now 👍